pkg/cidata/cidata.TEMPLATE.d/boot/30-install-packages.sh

```diff
@@ -1,29 +1,86 @@
 #!/bin/sh
 set -eux
 
-update_fuse_conf() {
-	# Modify /etc/fuse.conf (/etc/fuse3.conf) to allow "-o allow_root"
-	if [ "${LIMA_CIDATA_MOUNTS}" -gt 0 ]; then
-		fuse_conf="/etc/fuse.conf"
-		if [ -e /etc/fuse3.conf ]; then
-			fuse_conf="/etc/fuse3.conf"
-		fi
-		if ! grep -q "^user_allow_other" "${fuse_conf}"; then
-			echo "user_allow_other" >>"${fuse_conf}"
-		fi
+. /etc/os-release
```

Comment:
I wonder if this code should be more robust against various conditions:

* What if `/etc/os-release` doesn't exist?

  E.g. use `[ -f /etc/os-release ] && source /etc/os-release` ?

* What if `ID` or `ID_LIKE` are not set?

  This would throw an error because we have `set -u` at the top. Maybe preset the variables to the empty string before sourcing `/etc/os-release`?

* What if `ID_LIKE` lists multiple ids (e.g. `rhel fedora` instead of just `fedora`?

  Maybe do some regular expression match? `/bin/sh` doesn't support this though, and we can't rely on `bash`. Can we rely on `expr` being available, so we can do something like `expr "$ID_LIKE" : ".*fedora" >/dev/null`?

Reply:
You are absolutely right the code should be made more robust. I implemented all of your suggestions.

As for the availability of `expr` we should be able to rely on that since it [seems to be part of the standard](https://pubs.opengroup.org/onlinepubs/9699919799/)

Let me know if you think there is anything else I should add/remove/rework.

Reply:
Please fix `shellcheck` errors and remove trailing spaces from commands:

```diff
--- pkg/cidata/cidata.TEMPLATE.d/boot/30-install-packages.sh
+++ pkg/cidata/cidata.TEMPLATE.d/boot/30-install-packages.sh
@@ -6,6 +6,7 @@ DISTRIBUTION=""
 INSTALL_IPTABLES=0
 ID_LIKE=""
 ID=""
+# shellcheck source=/dev/null
 [ -f /etc/os-release ] && . /etc/os-release


@@ -15,22 +16,22 @@ main() {
        install_minimal_dependencies
        setup_dns
        # update_fuse_conf has to be called after installing all the packages,
-       # otherwise apt-get fails with conflict
-       update_fuse_conf
+       # otherwise apt-get fails with conflict
+       update_fuse_conf
 }

 determine_distribution() {
-       if command -v apt-get >/dev/null 2>&1 && [ $(expr "$ID_LIKE" : ".*debian") -gt 0 ]; then
+       if command -v apt-get >/dev/null 2>&1 && [ "$(expr "$ID_LIKE" : ".*debian")" -gt 0 ]; then
                DISTRIBUTION="debian_like"
-       elif command -v dnf >/dev/null 2>&1 && [ $(expr "$ID" : ".*fedora") -gt 0 ]; then
+       elif command -v dnf >/dev/null 2>&1 && [ "$(expr "$ID" : ".*fedora")" -gt 0 ]; then
                DISTRIBUTION="redhat_like"
-       elif command -v yum >/dev/null 2>&1 && [ $(expr "$ID_LIKE" : ".*centos") -gt 0 ]; then
+       elif command -v yum >/dev/null 2>&1 && [ "$(expr "$ID_LIKE" : ".*centos")" -gt 0 ]; then
                DISTRIBUTION="centos_like"
-       elif command -v pacman >/dev/null 2>&1 && [ $(expr "$ID_LIKE" : ".*arch") -gt 0 ]; then
+       elif command -v pacman >/dev/null 2>&1 && [ "$(expr "$ID_LIKE" : ".*arch")" -gt 0 ]; then
                DISTRIBUTION="arch_like"
-       elif command -v zypper >/dev/null 2>&1 && [ $(expr "$ID_LIKE" : ".*suse") -gt 0 ]; then
+       elif command -v zypper >/dev/null 2>&1 && [ "$(expr "$ID_LIKE" : ".*suse")" -gt 0 ]; then
                DISTRIBUTION="suse_like"
-       elif command -v apk >/dev/null 2>&1 && [ $(expr "$ID" : ".*alpine") -gt 0 ]; then
+       elif command -v apk >/dev/null 2>&1 && [ "$(expr "$ID" : ".*alpine")" -gt 0 ]; then
                DISTRIBUTION="alpine_like"
        else
                guess_distribution_based_on_package_manager
```

(the diff above doesn't show the trailing spaces correctly, but you can see the 2 lines that have them).